File: 1127_DataCouch_AzureOpenAI_S6/src/Java/PalindromeChecker.java
// PalindromeChecker.java
public class PalindromeChecker {

    public static boolean isPalindrome(String s) {
        if (s == null) {
            return false;
        }

        int left = 0;
        int right = s.length() - 1;

        while (left < right) {
            if (s.charAt(left) != s.charAt(right)) {
                System.out.println("The elements aren't the same, hence we return false");
                return false;
            }

            left++;
            right--;
        }

        return true;
    }
}

File: 1206_DataCouch_AzureOpenAI_S7/src/JS/palindromeChecker.js
// palindromeChecker.js
function isPalindrome(s) {
    if (s === null || s === "" || s === undefined) {
        return false;
    }

    let left = 0,
        right = s.length - 1;

    while (left < right) {
        console.log(`Current elements left index: '${s[left]}' and right index: '${s[right]}'`);

        if (s[left] != s[right]) {
            console.log("The elements aren't the same, hence we return false");
            return false;
        }

        left++;
        right--;
    }

    return true;
}

module.exports = { isPalindrome };
